Mage/src/main/java/mage/abilities/effects/common/combat/CantBeBlockedByOneAllEffect.java

```diff
@@ -49,24 +55,12 @@ public CantBeBlockedByOneAllEffect copy() {
     }
 
     @Override
-    public boolean apply(Layer layer, SubLayer sublayer, Ability source, Game game) {
-        switch (layer) {
-            case RulesEffects:
-                for (Permanent perm: game.getBattlefield().getActivePermanents(filter, source.getControllerId(), source, game)) {
-                    perm.setMinBlockedBy(amount);
-                }
-                break;
-            }
-        return true;
-    }
-
-    @Override
-    public boolean apply(Game game, Ability source) {
-        return false;
-    }
+    public boolean applies(Permanent permanent, Ability source, Game game) {
+        if (permanent == null || !filter.match(permanent, source.getControllerId(), source, game)) {
+            return false;
+        }
 
-    @Override
-    public boolean hasLayer(Layer layer) {
-        return layer == Layer.RulesEffects;
+        permanent.setMinBlockedBy(amount);
```

Comment:
`Applies` for read only operations and can be called multiple times from any place, do not change game data in that methods. Use `apply/replace` instead (for game state changes and choose dialogs).

Reply:
See my comment in ContinuousEffects.java `applies` are now properly readonly, and `apply` is called once on the RulesEffect layer.